shpc/main/modules/base.py

```diff
@@ -440,3 +461,58 @@ def view_install(self, view_name, name, force=False, container_image=None):
         # Don't continue if it exists, unless force is True
         view.confirm_install(module.module_dir, force=force)
         view.install(module.module_dir)
+
+    def reinstall(self, name, when_missing=None):
+        """
+        Reinstall the module, or all modules
+        """
+        if name:
+            module_name, _, version = name.partition(":")
+            # Find all the versions currently installed
+            installed_modules = self._get_module_lookup(
+                self.settings.module_base, self.modulefile, module_name
+            )
+            if (module_name not in installed_modules) or (
+                version and version not in installed_modules[module_name]
+            ):
+                logger.exit("%s is not installed. Nothing to reinstall." % name)
+            versions = [version] if version else installed_modules[module_name]
+            # Reinstall the required version(s) one by one
+            for version in versions:
+                self._reinstall(module_name, version, when_missing)
+        else:
+            # Reinstall everything that is currently installed
+            installed_modules = self._get_module_lookup(
```

Comment:
Hmm I wonder if there is some way to simplify the logic here (and not call the function in both blocks but use a shared call?) See duplicate of (almost) identical code always makes me pause!